generate_colors.rb

```diff
@@ -2,6 +2,43 @@
 require 'json'
 
 class ColorGenerator
+  RGBA = Struct.new(:r, :g, :b, :a) do
+    def self.from_hex(val)
+      val = val[1..-1] if val.start_with?("#")
+      chunks = val.scan(/.{2}/).map { |num| num.to_i(16) }
+      chunks.push(255) if chunks.size == 3
+      self.new(*chunks)
+    end
+
+    def join_weight(sep, round = 2)
```

Comment:
since this is a RGBA class -- does it make more sense to override `to_s` with `to_s(sep)`?

Reply:
My thought is to_s is ambiguous: join weights vs join ints vs hex